autobot-backend/testkit/auth_middleware_stub.py

```
@@ -0,0 +1,187 @@
+    # async and hand out an ``AsyncMock`` for every
+    # ``@patch("...get_current_user")`` site (patch() has done this since
+    # Python 3.8), so ``mock.return_value = {...}`` keeps working unchanged
+    # everywhere it's used today. The ONE caller that is NOT ``patch()``-based
+    # -- ``tests/integration/test_execution_snapshot_api.py``'s FastAPI
+    # ``dependency_overrides`` delegate -- could not stay a bare sync lambda
+    # calling this coroutine function without awaiting it; it was updated in
+    # lockstep in the same commit to await it.
+    # The ``request`` parameter is annotated ``Request`` so FastAPI injects it
+    # instead of treating it as a request field, and defaults to None so
+    # direct ``get_current_user()`` call sites keep working.
+    async def _get_current_user_stub(request: _FastAPIRequest = None) -> dict:  # type: ignore[assignment]
```

Ruff (0.16.3)

[warning] Unused function argument: `request`

(ARG001)